packages/mdc-ripple/foundation.js

```diff
@@ -404,7 +418,10 @@ class MDCRippleFoundation extends MDCFoundation {
       }
 
       if (needsActualDeactivation) {
-        this.activationState_ = this.defaultActivationState_();
+        // Prevents double-animation when 'mouse' and 'touch' events fire sequentially
```

Comment:
Yes. The double-animation happens because of the 408-410 change. The reason for the 408-410 change is because when touching an element, mousedown, pointerdown, and touchstart all fire, but if the finger slides off the element before being lifted only touchend fires, but mouseup and pointerup never get fired (I didn't check the specs but I wonder if it's a browser bug). As a result, in such a case the ripple won't be deactivated by a mouseup so we need to check for a touchend as well. The problem is that then in a normal touch scenario (when the finger doesn't move) we would get a double activation/deactivation for both the mouse and touch events. That's why I added a setTimeout to delay the deactivation until after the animation period, which in turn prevents the second event from activating (since we don't allow an active ripple to be activated again).

Reply:
Thanks for the information. This raises another concern, however, now that you explained this: this means that ripples no longer re-trigger if they are legitimately re-activated while the previous ripple is still animating. (E.g., click or keydown on a button twice relatively fast; in the existing implementation, the ripple will restart, but due to this fix it will no longer do that, which may make the component seem unresponsive, since it in fact _was_ activated again.)

Reply:
Interesting - I didn't realize that. What if we only allow a double animation if the second animation is fired by the same event (e.g. 'mousedown + mousedown' as opposed to 'mousedown + touchstart')? I think it would be rare that someone would use two different methods of activation in such a quick timeframe, no?

Reply:
That thought occurred to me; I'm also wondering if it'd be feasible to use a tighter timeout to avoid this specific issue? i.e. when the sequential events occur due to one user input, do they actually occur nearly simultaneously, or is this a case where there's a tap delay?

Reply:
I would need to test more thoroughly. When I lowered the delay to 100ms I think it still happened, but I'll check again and try to nail down the delay range.